### ISSUE TITLE
pglast>-6 install from wheel is missing symbol

### ISSUE THREAD

**Issue by declaresub** [lelit/pglast#144]
After installing pglast 6.1 from a wheel via pip, I was unable to import the package.

```
>>> import pglast
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/charles/Documents/git/wxtg/cynque-api-server/.venv/3.12.0/lib/python3.12/site-packages/pglast/__init__.py", line 14, in <module>
    from .parser import fingerprint, get_postgresql_version, parse_sql, scan, split
ImportError: dlopen(/Users/charles/Documents/git/wxtg/cynque-api-server/.venv/3.12.0/lib/python3.12/site-packages/pglast/parser.cpython-312-darwin.so, 0x0002): symbol not found in flat namespace '_pg_query__keyword_kind__descriptor'
```

This happened on my MacBook Pro, Apple M1 Max, Mac OS 14.2, and python 3.12.

I was able to build, install, and import pglast 6.1 from source.

**Comment by lelit** [comment]
I will try to investigate, but since Apple machines are a scarce resource around me it's a bit tricky.

Am I right assuming this does not affect "old" Intel-based Macs?

**Comment by lelit** [comment]
Indeed, the wheel built for Apple Silicon macs is b0rked: it seems that the extension module (`pglast/parser.cpython-311-darwin.so`) was not properly linked against the static `libpg_query.a`... and in fact it is a mere 606Kb, confronted with the 3.7Mb of the same shared object produced for `x86_64`... A local compile of the module on a macpro m3 produces a more reasonable and working 4.4Mb `.so` file.
I really don't know what's going on here :sob:

**Comment by lelit** [comment]
@declaresub: I tried addressing this issue in the [homonym branch](https://github.com/lelit/pglast/tree/issue144); any chance you could try installing the wheel produced [here](https://github.com/lelit/pglast/actions/runs/7670977433) and see if it works as expected?

**Comment by declaresub** [comment]
Sure.  Where is the wheel package?  I'm nit finding it.

**Comment by declaresub** [comment]
I installed the wheel and got the same error, unfortunately.

**Comment by lelit** [comment]
> I installed the wheel and got the same error, unfortunately.

Ok, thank for trying. I will ask around.

**Comment by lelit** [comment]
I made another experiment, that apparently produced _normal sized_ [wheels](https://github.com/lelit/pglast/actions/runs/7680899789/artifacts/1199925084)... as the [changelog says](https://github.com/lelit/pglast/commit/877e07c8ff5166705d0b261a4c50a9e5eaeb7de6) it's a bit extreme, but if it works...

**Comment by lelit** [comment]
Update: AFAICT, that last build produced a working wheel also for ARM64.

**Comment by lelit** [comment]
@declaresub, a double check would be appreciated :wink:

**Comment by declaresub** [comment]
I just double-checked with success -- installed from wheel in a fresh virtual environment, and successfully imported pglast.

**Comment by lelit** [comment]
GH [yesterday announced](https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/) a new macOS M1 runner, that apparently produced [correct wheels](https://github.com/lelit/pglast/actions/runs/7722805715/artifacts/1208024991).

**Comment by lelit** [comment]
I just released v6.2, and I'm confident it fixes this problem.
